rules/prefer-dom-node-dataset.js

```diff
@@ -1,34 +1,40 @@
 'use strict';
 const {isIdentifierName} = require('@babel/helper-validator-identifier');
-const escapeString = require('./utils/escape-string.js');
-const {methodCallSelector, matches} = require('./selectors/index.js');
+const {escapeString} = require('./utils/index.js');
+const {isMethodCall, isStringLiteral} = require('./ast/index.js');
 
 const MESSAGE_ID = 'prefer-dom-node-dataset';
 const messages = {
 	[MESSAGE_ID]: 'Prefer `.dataset` over `{{method}}(…)`.',
 };
 
-const selector = [
-	matches([
-		methodCallSelector({method: 'setAttribute', argumentsLength: 2}),
-		methodCallSelector({methods: ['getAttribute', 'removeAttribute', 'hasAttribute'], argumentsLength: 1}),
-	]),
-	'[arguments.0.type="Literal"]',
-].join('');
-
 const dashToCamelCase = string => string.replace(/-[a-z]/g, s => s[1].toUpperCase());
 
 /** @param {import('eslint').Rule.RuleContext} context */
 const create = context => ({
-	[selector](node) {
-		const [nameNode] = node.arguments;
-		let attributeName = nameNode.value;
-
-		if (typeof attributeName !== 'string') {
+	CallExpression(node) {
+		if (!(
+			(
+				isMethodCall({
+					method: 'setAttribute',
+					argumentsLength: 2,
+					optionalCall: false,
+					optionalMember: false,
+				})
+				|| isMethodCall({
+					methods: ['getAttribute', 'removeAttribute', 'hasAttribute'],
+					argumentsLength: 1
+					optionalCall: false,
```

Comment:
Just curious, why not just make this the default?

Reply:
What if we want ignore `optionalCall` check

```js
isMethodCall({
	optionalCall: undefined
})
```

? Odd to me.

Reply:
Maybe introduce another flag?

```js
isMethodCall(node, {
	checkOptionalCall: false
})
```